independent-projects/tools/common/src/main/java/io/quarkus/devtools/commands/handlers/QuarkusCommandHandlers.java

```diff
@@ -11,111 +17,86 @@
+                if (result.matches()) {
+                    final Set<AppArtifactCoords> withStrippedVersion = result.getExtensions().stream().map(Extensions::toCoords)
+                            .map(Extensions::stripVersion).collect(Collectors.toSet());
+                    // We strip the version because those extensions are managed
+                    builder.addAll(withStrippedVersion);
                 } else {
-                    SelectionResult result = select(query, registry, false);
-                    if (!result.matches()) {
-                        StringBuilder sb = new StringBuilder();
-                        // We have 3 cases, we can still have a single candidate, but the match is on label
-                        // or we have several candidates, or none
-                        Set<Extension> candidates = result.getExtensions();
-                        if (candidates.isEmpty()) {
-                            // No matches at all.
-                            PRINTER.nok(" Cannot find a dependency matching '" + query + "', maybe a typo?");
-                            success = false;
-                        } else {
-                            sb.append(Printer.NOK).append(" Multiple extensions matching '").append(query).append("'");
-                            result.getExtensions()
-                                    .forEach(extension -> sb.append(System.lineSeparator()).append("     * ")
-                                            .append(extension.managementKey()));
-                            sb.append(System.lineSeparator())
-                                    .append("     Be more specific e.g using the exact name or the full GAV.");
-                            PRINTER.print(sb.toString());
-                            success = false;
-                        }
-                    } else { // Matches.
-                        for (Extension extension : result) {
-                            // Don't set success to false even if the dependency is not added; as it's should be idempotent.
-                            updated = buildFile.addDependency(invocation.getPlatformDescriptor(), extension) || updated;
-                        }
+                    StringBuilder sb = new StringBuilder();
+                    // We have 3 cases, we can still have a single candidate, but the match is on label
+                    // or we have several candidates, or none
+                    Set<Extension> candidates = result.getExtensions();
+                    if (candidates.isEmpty()) {
+                        // No matches at all.
+                        invocation.log().info(nok(" Cannot find a dependency matching '" + query + "', maybe a typo?"));
+                        return null;
+                    } else {
+                        sb.append(NOK).append(" Multiple extensions matching '").append(query).append("'");
```

Comment:
Since you're refactoring `Printer` into `ConsoleMessageFormat`, try to not expose the `NOK` attribute there otherwise the methods in that class are meaningless